File: Programming Basics with Python/По-сложни проверки/fruit_shop.py

product = input()
day = input()
quantity = float(input())

total_price = 0
if day == "Monday" \
    or day == "Tuesday" \
    or day == "Wednesday" \
    or day == "Thursday" \
    or day == "Friday":
    if product == "banana":
        total_price = quantity * 2.50
    elif product == "apple":
        total_price = quantity * 1.20
    elif product == "orange":
        total_price = quantity * 0.85
    elif product == "grapefruit":
        total_price = quantity * 1.45
    elif product == "kiwi":
        total_price = quantity * 2.70
    elif product == "pineapple":
        total_price = quantity * 5.50
    elif product == "grapes":
        total_price = quantity * 3.85
    else:
        print("error")
elif day == "Saturday" \
    or day == "Sunday":
    if product == "banana":
        total_price = quantity * 2.70
    elif product == "apple":
        total_price = quantity * 1.25
    elif product == "orange":
        total_price = quantity * 0.90
    elif product == "grapefruit":
        total_price = quantity * 1.60
    elif product == "kiwi":
        total_price = quantity * 3.00
    elif product == "pineapple":
        total_price = quantity * 5.60
    elif product == "grapes":
        total_price = quantity * 4.20
    else:
        print("error")
else:
    print("error")

if total_price != 0:
    print(f"{total_price:.2f}")

File: Programming Basics with Java/More complex checks/FruitShop.java
import java.util.Scanner;

public class FruitShop 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String fruit = input.nextLine();
        String day = input.nextLine();
        double quantity = input.nextDouble();

        double price = 0.0;

        switch (day)
        {
            case "Monday":
            case "Tuesday":
            case "Wednesday":
            case "Thursday":
            case "Friday":
            {
                switch (fruit)
                {
                    case "banana":
                    {
                        price = 2.5;
                        break;
                    }
                    case "apple":
                    {
                        price = 1.2;
                        break;
                    }
                    case "orange":
                    {
                        price = 0.85;
                        break;
                    }
                    case "grapefruit":
                    {
                        price = 1.45;
                        break;
                    }
                    case "kiwi":
                    {
                        price = 2.7;
                        break;
                    }
                    case "pineapple":
                    {
                        price = 5.5;
                        break;
                    }
                    case "grapes":
                    {
                        price = 3.85;
                        break;
                    }
                }
                break;
            }
            case "Saturday":
            case "Sunday":
            {
                switch (fruit)
                {
                    case "banana":
                    {
                        price = 2.7;
                        break;
                    }
                    case "apple":
                    {
                        price = 1.25;
                        break;
                    }
                    case "orange":
                    {
                        price = 0.9;
                        break;
                    }
                    case "grapefruit":
                    {
                        price = 1.6;
                        break;
                    }
                    case "kiwi":
                    {
                        price = 3;
                        break;
                    }
                    case "pineapple":
                    {
                        price = 5.6;
                        break;
                    }
                    case "grapes":
                    {
                        price = 4.2;
                        break;
                    }
                }
                break;
            }
        }

        if(price != 0)
        {
            System.out.printf("%.2f" ,price * quantity);
        }
        else
        {
            System.out.println("error");
        }
    }
}
